From bea87d2cde7c490c8417a97f043eaa6905c27f08 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Wed, 3 Jun 2020 16:38:49 +0200
Subject: [PATCH] [cmake] Fix ROOT/RLogger.hxx not found:

When building without root7 but with C++14, RLogger is used.
Make it available to the include directories.

(cherry picked from commit fbe075ceedcd7d7f9993864c82ac5bd2960f3494)
---
 core/base/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/base/CMakeLists.txt b/core/base/CMakeLists.txt
index 66ec9a2224..95eea9b44a 100644
--- a/core/base/CMakeLists.txt
+++ b/core/base/CMakeLists.txt
@@ -198,6 +198,7 @@ set(BASE_SOURCES
 if(CMAKE_CXX_STANDARD GREATER 11)
   list(APPEND BASE_HEADERS ROOT/RLogger.hxx)
   list(APPEND BASE_SOURCES v7/src/RLogger.cxx)
+  set(BASE_V7_INC ${CMAKE_SOURCE_DIR}/core/base/v7/inc)
   # TLogger.hxx may be used without root7 flag, but is placed in v7/inc/,
   # so we need to tell ROOT_INSTALL_HEADERS() where to find it
   set(BASE_HEADER_DIRS inc/ v7/inc/)
@@ -220,6 +221,7 @@ set(Core_dict_headers ${BASE_HEADERS} PARENT_SCOPE)
 ROOT_OBJECT_LIBRARY(Base ${BASE_SOURCES})
 
 target_include_directories(Base PRIVATE
+   ${BASE_V7_INC}
    ${PCRE_INCLUDE_DIR}
    res
    ${CMAKE_SOURCE_DIR}/core/foundation/res
-- 
2.27.0

